src/Tool-CriticBrowser/CBSelectBrowser.class.st

```diff
@@ -0,0 +1,134 @@
+"
+I'm an abstract SpPresenter
+
+my subclass have to define 
+	addAllItems
+	addItems 
+	nextAction
+	removeAllItems
+	removeItems 
+
+I define 
+	label of button
+	 
+    Instance Variables
+	addAllButton:		<SPButtonPresenter>
+	addButton:		<SPButtonPresenter>
+	nextButton:		<SPButtonPresenter>
+	removeAllButton:		<SPButtonPresenter>
+	removeButton:		<SPButtonPresenter>
+	
+	selectedPresenter:		<Object>
+	toSelectPresenter:		<Object>
+			those 2 instance Variable will 
+"
+Class {
+	#name : #CBSelectBrowser,
+	#superclass : #SpPresenter,
+	#instVars : [
+		'selectedPresenter',
+		'toSelectPresenter',
+		'addAllButton',
+		'addButton',
+		'nextButton',
+		'removeAllButton',
+		'removeButton'
+	],
+	#category : #'Tool-CriticBrowser-Parts'
+}
+
+{ #category : #specs }
+CBSelectBrowser class >> defaultSpec [
+	^ SpBoxLayout newVertical
+		add:
+			(SpPanedLayout newHorizontal
+				add: #toSelectPresenter;
+				add: #selectedPresenter;
+				yourself);
+		add:
+			(SpBoxLayout newHorizontal
+				add: #addButton;
+				add: #addAllButton;
+				add: #removeButton;
+				add: #removeAllButton;
+				add: #nextButton;
+				yourself)
+			withConstraints: [ :contraint | contraint height: 25 ];
```

Comment:
this approach is incorrect.
To create action bars as this forces the size of the component and then is hard to scale later. 
There is already a presenter made to contain action buttons: `SpActionBarPresenter`, which receive buttons as desired: 
```Smalltalk
actionBar := self newActionBar
		add: (self newButton
			label: 'Add'; 
			icon: (self iconNamed: #add);
			help: 'Add a new element to the list.';
			action: [ self doAdd ];
			yourself);
		yourself```

and the advantage is that we can style it later.

Reply:
and YES, I know documentation is still lacking ;)

Reply:
I'm working on it Thanks you! it will be in the next commit